app/uk/gov/hmrc/gform/formtemplate/FormComponentMaker.scala

```diff
@@ -62,6 +62,14 @@ class FormComponentMaker(json: JsValue) {
     case JsUndefined() => Right(EmailVerification.noVerification)
   }
 
+  lazy val optFileUploadProvider: Opt[FileUploadProvider] = (json \ "service") match {
+    case JsDefined(JsString("upscan"))     => Right(FileUploadProvider.Upscan)
+    case JsDefined(JsString("fileUpload")) => Right(FileUploadProvider.FileUploadFrontend)
+    case JsDefined(unknown) =>
+      Left(UnexpectedState(s"Unsupported file upload service '$unknown'. Only 'upscan' or 'fileUpload' are supported"))
+    case JsUndefined() => Right(FileUploadProvider.FileUploadFrontend)
+  }
+
```

Comment:
Minor: we can avoid usage of  JsDefine, JsString etc

```
  lazy val optFileUploadProvider1 = (json \ "service").asOpt[String].fold[Opt[FileUploadProvider]](Right(FileUploadProvider.FileUploadFrontend)) {
    case "upscan" => Right(FileUploadProvider.Upscan)
    case "fileUpload" => Right(FileUploadProvider.FileUploadFrontend)
    case unknown => Left(UnexpectedState(s"Unsupported file upload service '$unknown'. Only 'upscan' or 'fileUpload' are supported"))
  }
```

Reply:
This definition is not the same and when `(json \ "service")` is not a String it pass, instead of warning you.